db/collab.go

```diff
@@ -53,19 +61,69 @@ func (s *Session) RemoveConn(uid string) error {
 		return errors.New("Could not remove unconnected user")
 	}
 	delete(s.Conns, uid)
+	// Stop other connections from sending to removed connection
+	for _, conn := range s.Conns {
+		delete(conn.Subscriptions, uid)
+	}
 	return nil
 }
 
-func (s *Session) Broadcast(msg Message) error {
-	var mostRecentError error
+func (s *Session) BroadcastAll(msg Message) (lastErr error) {
 	s.Lock.Lock()
 	defer s.Lock.Unlock()
 	for _, conn := range s.Conns {
-		if err := websocket.JSON.Send(conn, msg); err != nil {
-			mostRecentError = err
+		if err := websocket.JSON.Send(conn.Conn, msg); err != nil {
+			lastErr = err
+		}
+	}
+	return
+}
+
+func (s *Session) BroadcastError(uid string, err string) error {
+	errorMsg := Message{
+		Author: uid,
+		Type:   "ERROR",
+		Body:   err,
+	}
+	if broadcastErr := s.BroadcastTo(errorMsg, uid); broadcastErr != nil {
+		return broadcastErr
+	}
+	return nil
+}
+
+func (s *Session) BroadcastTo(msg Message, uids ...string) (lastErr error) {
```

Comment:
While wrapping up documentation, two short doc comments on `BroadcastTo` and `BroadcastToSet` will help.